nursery/import-public-key.yml

```diff
@@ -0,0 +1,14 @@
+rule:
+  meta:
+    name: import public key
+    namespace: data-manipulation/encryption
+    author: william.ballenthin@fireeye.com
+    scope: function
+    examples:
+      # - unpacked Cl0p ransomware
+  features:
+    - and:
+      - api: crypt32.CryptStringToBinary
+      - api: crypt32.CryptDecodeObjectEx
+      - api: crypt32.CryptAcquireContext
+      - api: crypt32.CryptImportPublicKeyInfo

```

Comment:
are all of these APIs necessary or can we loosen it up to catch similar samples?

Reply:
I looked up a reference sample in an old copy of cryptolocker (`ffeae4a391a1d5203bd04b4161557227`).  The functionality of the rule was broken up into 2 functions.

The first function @ `00404c20` will call `CryptStringToBinary` and then `CryptDecodeObjectEx`.  
The second function @ `004047a0` will call `CryptAcquireContext` then ` CryptImportPublicKeyInfo`

Breaking them up to help trigger on this variant may lose the original intent of the rule.  

`CryptImportPublicKeyInfo` seems to be the primary functionality that the rule is triggering on. So we could probably reduce the rule more to be:

```
      - and:
         - api: crypt32.CryptAcquireContext
         - api: crypt32.CryptImportPublicKeyInfo
         - optional: 
             - and:
                - api: crypt32.CryptDecodeObjectEx
                - api: crypt32.CryptStringToBinary
```

The other functionality of `CryptStringToBinary` would be nice to have as an optional indicator though as it provides nice insight into the behavior.

Reply:
i like the suggestions here a lot.

since i was working off a single example (the malware i was currently reversing) i wasn't sure how the rule generalized. the updates make a lot of sense. i'll fix this PR.

Reply:
added in 36bbb25